src/perform_step/extrapolation_perform_step.jl

```diff
@@ -1693,41 +1693,132 @@ function perform_step!(integrator, cache::ImplicitHairerWannerExtrapolationCache
+                  return
+                end
+              end
+              @.. diff1[Threads.threadid()] = u_temp3[Threads.threadid()] - u_temp4[Threads.threadid()]
+            end
+          end
+        end
+      end
+    else
+      let n_curr=n_curr,subdividing_sequence=subdividing_sequence,uprev=uprev,dt=dt,u_temp3=u_temp3,
+          u_temp4=u_temp4,k_tmps=k_tmps,p=p,t=t,T=T
+        Threads.@threads for i in 0:(n_curr ÷ 2)
+          indices = i != n_curr - i ? (i, n_curr - i) : (n_curr-i) #Avoid duplicate entry in tuple
+          for index in indices
+            j_int_temp = 4 * subdividing_sequence[index+1]
+            dt_int_temp = dt / j_int_temp # Stepsize of the ith internal discretisation
+            jacobian2W!(W[Threads.threadid()], integrator.f.mass_matrix, dt_int_temp, J, false)
+            @.. u_temp4[Threads.threadid()] = uprev
+            @.. linsolve_tmps[Threads.threadid()] = dt_int_temp * fsalfirst
+            cache.linsolve[Threads.threadid()](vec(k_tmps[Threads.threadid()]), W[Threads.threadid()], vec(linsolve_tmps[Threads.threadid()]), !repeat_step)
+            @.. k_tmps[Threads.threadid()] = -k_tmps[Threads.threadid()]
+            @.. u_temp3[Threads.threadid()] = u_temp4[Threads.threadid()] + k_tmps[Threads.threadid()] # Euler starting step
+            @.. diff1[Threads.threadid()] = u_temp3[Threads.threadid()] - u_temp4[Threads.threadid()]
+            for j in 2:j_int_temp + 1
+              f(k_tmps[Threads.threadid()], cache.u_temp3[Threads.threadid()], p, t + (j-1) * dt_int_temp)
+              @.. linsolve_tmps[Threads.threadid()] = dt_int_temp*k_tmps[Threads.threadid()] - (u_temp3[Threads.threadid()] - u_temp4[Threads.threadid()])
+              cache.linsolve[Threads.threadid()](vec(k_tmps[Threads.threadid()]), W[Threads.threadid()], vec(linsolve_tmps[Threads.threadid()]), !repeat_step)
+              @.. k_tmps[Threads.threadid()] = -k_tmps[Threads.threadid()]
+              @.. T[index+1] = 2*u_temp3[Threads.threadid()] - u_temp4[Threads.threadid()] + 2*k_tmps[Threads.threadid()] # Explicit Midpoint rule
+              if(j == j_int_temp + 1)
+                @.. T[index + 1] = 0.5(T[index + 1] + u_temp4[Threads.threadid()])
+              end
+              @.. u_temp4[Threads.threadid()] = u_temp3[Threads.threadid()]
+              @.. u_temp3[Threads.threadid()] = T[index+1]
+              if(index<=1)
+                # Deuflhard Stability check for initial two sequences 
+                @.. diff2[Threads.threadid()] = u_temp3[Threads.threadid()] - u_temp4[Threads.threadid()]
+                if(integrator.opts.internalnorm(diff1[Threads.threadid()],t)<integrator.opts.internalnorm(0.5*(diff2[Threads.threadid()] - diff1[Threads.threadid()]),t))
+                  # Divergence of iteration, overflow is possible. Force fail and start with smaller step
+                  integrator.force_stepfail = true
+                  return
+                end
```

Comment:
It seems to me that this block might be the culprit because now we have enabled multi-threading, how to check for stability correctly?

Reply:
there should be a switch, and if one thread goes unstable they should all abort. It's okay if some start to do their next computation, as long as then they all then sync up after

Reply:
Okay let me try to implement it

Reply:
I have fixed with a switch which breaks the threading loop:
Results:
```
julia> sol = solve(prob,ImplicitHairerWannerExtrapolation(threading=true))
retcode: Success
Interpolation: 3rd order Hermite
t: 62-element Array{Float64,1}:
      0.0
      0.0015609382820287144
      0.005825835783645151
      0.01567575045726786
      0.16947966893502833
      1.743871372983346
      9.306194996461432
     31.770571909730194
     99.75191412340172
    251.80711169599098
    536.8233987024881
   1047.4098500692314
   1277.3618197236628
      ⋮
  43133.7921812233
  44533.0199685061
  50055.47993946205
  50143.549394361304
  50645.7867089978
  53200.886884522
  67639.73166022974
  67726.34146826199
  68290.34600170281
  71327.41288918354
  88525.6423629744
 100000.0
u: 62-element Array{Array{Float64,1},1}:
 [1.0, 0.0, 0.0]
 [0.9999375686612398, 3.419788713381114e-5, 2.8233451626510687e-5]
 [0.9997671708792938, 3.6480839574421016e-5, 0.0001963617819497166]
 [0.9993747529950145, 3.6408757655733936e-5, 0.0005888518373725846]
 [0.9934418204907772, 3.532707974034257e-5, 0.006522860595101827]
 [0.9473878487210842, 2.7843504922838892e-5, 0.05258355109327624]
 [0.8469989043693693, 1.6688688600730842e-5, 0.152974647487413]
 [0.7387518670436833, 1.0136165147203725e-5, 0.26123561998272277]
 [0.6174763123317458, 6.160142276754425e-6, 0.38247879318679445]
 [0.5076171597118828, 4.028549430990326e-6, 0.4920810102753788]
 [0.4131543327232221, 2.7803011885121084e-6, 0.5858827290916845]
 [0.3290933691209731, 1.940403291877344e-6, 0.6680100908138078]
 [0.3045872886094369, 1.7328634339264573e-6, 0.6916504292086891]
 ⋮
 [0.036616402611313, 1.537768134569056e-7, 0.9519856200593746]
 [0.0356457232493685, 1.495630266061263e-7, 0.9529511700095585]
 [0.0322879960671094, 1.348414844453152e-7, 0.95628613653051]
 [0.03223986774524779, 1.3479187702398253e-7, 0.9563342193567873]
 [0.031969297577572324, 1.336219095630056e-7, 0.9566055821982409]
 [0.030652705785227934, 1.2794923477669555e-7, 0.957916211258885]
 [0.02521614608213966, 1.0470432195780831e-7, 0.9636204815970087]
 [0.025187746750192412, 1.0451690725493389e-7, 0.9636488960956322]
 [0.025003614528729616, 1.0373405338488132e-7, 0.9638323513852816]
 [0.02405705910224919, 9.971021115537788e-8, 0.9647748199881884]
 [0.019750460767379453, 8.151428218922708e-8, 0.9689778794510242]
 [0.017683441147082677, 7.276944652347233e-8, 0.9710151320645249]

julia> versioninfo()
Julia Version 1.4.1
Commit 381693d3df* (2020-04-14 17:20 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 6
```

Reply:
what was the main cause of the issue before?